earthpy/tests/test_plot.py

```diff
@@ -28,27 +27,27 @@ def test_num_titles():
     should raise an error OR if the title list is a different length than
     the array it should also raise an errors"""
 
-    single_band = im[0]
+    single_band = im_arr[0]
```

Comment:
Here `im_arr` is being pulled from the environment outside of the function implicitly, which can make it harder to understand the test. If we used a fixture for `im_arr` (e.g., the `image_array_3d` fixture suggested above), then the test could be rewritten in a way that makes it explicit what objects are required for the test to run: 

```python
def test_num_titles(image_array_3d):
    """If a user provides two titles for a single band array, the function
    should raise an error OR if the title list is a different length than
    the array it should also raise an errors"""
    single_band = image_array_3d[0]

    with pytest.raises(ValueError):
        es.plot_bands(arr=single_band,
                      title=["Title1", "Title2"])
    with pytest.raises(ValueError):
        es.plot_bands(arr=image_array_3d,
                      title=["Title1", "Title2", "Title3"])
```